server/helper.ts

```diff
@@ -108,6 +108,31 @@ function getSearchkitRouter() {
         similars = [];
       }
 
+      if(source){
+        // Get related publications from all other indices
+        const relPublValues = await elasticsearch.getStudyFieldAllIndices(req.params.id, 'relatedPublications', req.params.index);
+        const otherLangPublications = relPublValues.flatMap(item => {
+          return (item.values as RelatedPublication[] || []).map(publication => ({
+            title: publication.title,
+            holdings: publication.holdings,
+            lang: item.index ? item.index.split('_')[1] : ''
+          }));
+        });
+
+        const selectedLang = req.params.index.split('_')[1];
+        if(source.relatedPublications){
+          // Add lang to the related publications of selected language
+          source.relatedPublications.forEach(publication => {
+            publication.lang = selectedLang;
+          });
+          // Add related publications in other languages
+          source.relatedPublications.push(...otherLangPublications);
+        } else {
+          // Replace original relatedPublications with related publications found in other languages
+          source.relatedPublications = otherLangPublications;
+        }
+      }
```

Comment:
There is no handling of Elasticsearch errors so a failure of the second request leads to there being no response even if the first request succeeds.